hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/OperationExecutor.java

```diff
@@ -95,7 +99,15 @@
      * @throws java.lang.NullPointerException if op is null.
      * @throws IllegalThreadStateException    if the operation is not allowed to be run on the calling thread.
      */
-    void runOperationOnCallingThread(Operation op);
+    void runOnCallingThread(Operation op);
+
+    /**
+     * Tries to run the operation on the calling thread if possible. Otherwise offload it to a different thread.
```

Comment:
I don't want to use the works async and async here because it is too confusing. 

It means:
- run on calling thread 
- offload to different thread

You want to run on the calling thread as much as possible because it is faster. The logic used to be in the invocation, so the same logic is now moved into the scheduler to make the scheduler smarter and the invocation dumber.